src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -222,6 +228,54 @@ private boolean edgeCharacteristicsToIgnore(final Edge edge)
                 // Ignore edges that have been way sectioned at the border, as has high probability
                 // of creating a false positive due to the sectioning of the way
                 || SyntheticBoundaryNodeTag.isBoundaryNode(edge.end())
-                || SyntheticBoundaryNodeTag.isBoundaryNode(edge.start());
+                || SyntheticBoundaryNodeTag.isBoundaryNode(edge.start())
+                // Ignore edges that are of type service and is connected to pedestrian navigable
+                // ways
+                || this.isServiceRoad(edge) && this.isConnectedToPedestrian(edge);
+    }
+
+    /**
+     * Checks if the edge is fully enclosed within areas that have amenity tags that are in the
+     * AMENITY_VALUES_TO_EXCLUDE list
+     *
+     * @param edge
+     *            any edge
+     * @return true if the edge is fully enclosed within the area with excluded amenity tag
+     */
+    private boolean isWithinAreasWithExcludedAmenityTags(final Edge edge)
+    {
+        return StreamSupport
+                .stream(edge.getAtlas()
+                        .areasIntersecting(edge.bounds(),
+                                area -> Validators.isOfType(area, AmenityTag.class,
+                                        AMENITY_VALUES_TO_EXCLUDE))
+                        .spliterator(), false)
+                .anyMatch(area -> area.asPolygon().fullyGeometricallyEncloses(edge.asPolyLine()));
+    }
+
+    /**
+     * Checks if an {@link Edge} is of type service
+     * 
+     * @param edge
+     *            any edge
+     * @return true if the highway classification of the edge is of type service and it has a
+     *         service tag
+     */
+    private boolean isServiceRoad(final Edge edge)
+    {
+        return Validators.isOfType(edge, HighwayTag.class, HighwayTag.SERVICE)
+                && Validators.hasValuesFor(edge, ServiceTag.class);
+    }
+
+    /**
+     * Checks if an {@link Edge} is connected to any edge that is a pedestrian navigable highway
+     * 
+     * @param edge
+     *            any edge
+     * @return true if the edge has connection to pedestrian navigable highways
+     */
+    private boolean isConnectedToPedestrian(final Edge edge)
```

Comment:
`isConnectedToPedestrian` what? Network or highway?

Reply:
Have updated it to `isConnectedToPedestrianNavigableHighway`